phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/UngroupedAggregateRegionObserver.java

```diff
@@ -581,92 +601,135 @@ private boolean areMutationsInSameTable(Table targetHTable, Region region) {
                 region.getTableDescriptor().getTableName().getName()) == 0);
     }
 
+    @Override
+    public InternalScanner preFlush(ObserverContext<RegionCoprocessorEnvironment> c, Store store,
+            InternalScanner scanner, FlushLifeCycleTracker tracker) throws IOException {
+        if (!isPhoenixTableTTLEnabled(c.getEnvironment().getConfiguration())) {
+            return scanner;
+        } else {
+            return User.runAsLoginUser(new PrivilegedExceptionAction<InternalScanner>() {
+                @Override public InternalScanner run() throws Exception {
+                    String tableName = c.getEnvironment().getRegion().getRegionInfo().getTable()
+                            .getNameAsString();
+                    long maxLookbackInMillis =
+                            UngroupedAggregateRegionObserver.getMaxLookbackInMillis(
+                                    tableName, store.getColumnFamilyName(),
+                                    BaseScannerRegionObserverConstants.getMaxLookbackInMillis(
+                                            c.getEnvironment().getConfiguration()));
+                    maxLookbackInMillis = CompactionScanner.getMaxLookbackInMillis(tableName,
```

Comment:
Why are we maintaining two maps of max lookback age: one in this class and other in Compaction Scanner?

Reply:
The one in UngroupedAggregateRegionObserver is for caching the max lookback age in order to prevent getTable calls in the coproc flush hook. The other is for overriding the cached value by other coprocs, (for example a backup coproc).